continuous_integration/recipes/distributed/meta.yaml

```diff
@@ -24,28 +24,28 @@ build:
 
 requirements:
   host:
-    - python >=3.9
+    - python >=3.10
     - pip
     - dask-core {{ dask_version }}
     - versioneer =0.29
     - tomli # [py<311]
   run:
-    - python >=3.9
+    - python >=3.10
     - click >=8.0
-    - cloudpickle >=1.5.0
-    - cytoolz >=0.10.1
+    - cloudpickle >=2.0.0
```

Comment:
This does not match with the pyproject.toml dependencies. I'm fine with bumping this (according to pypi tags, the first cloudpickle version supporting 3.10 is actually 3.0)

If this goes away, we can also drop some compat code, see https://github.com/dask/distributed/blob/879e5924ccb4f1bb45e0090f6ca9e6f7eaf343ce/distributed/protocol/pickle.py#L13

Reply:
> according to pypi tags, the first cloudpickle version supporting 3.10 is actually 3.0

I went ahead and bumped the dependencies in `pyproject.toml` to have a minimum of 2.0.0. Looking through the git history, I think the PyPI tags were updated after 3.10 support was actually added. Also fine with bumping to 3.0.0 if folks prefer. 

> If this goes away, we can also drop some compat code, see

Dropped 👍